overlays/bootstrap.nix

```diff
@@ -497,7 +487,16 @@ in {
 
     # the bootstrap infra structure (pre-compiled ghc; bootstrapped cabal-install, ...)
     bootstrap = with final.haskell-nix;
-      let ghc = final.buildPackages.haskell-nix.bootstrap.compiler."${buildBootstrapper.compilerNixName}";
+      let
+        # This compiler-nix-name will only be used to build nix-tools and cabal-install
+        # when checking materialization of alex, happy and hscolour.
+        compiler-nix-name = "ghc865";
```

Comment:
Does this version have to match any of the other versions? When can we upgrade it?